source/contribute.rst

```diff
@@ -192,6 +195,70 @@ this in case you want to read more about this:
 - `Commit message style for git <https://commit.style/>`_
 - `A note about git commit messages <https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html>`_
 
+To squash or not to squash
+--------------------------
+
+Of course, if you want to craft good commit messages, you will have a
+hard time if the changeset you are describing does too many things. That
+might very well happen if you notice small things along the way that are
+unrelated to your PR, but too small to warrant a separate one. ``git add
+--patch`` or ``git add -p`` will be of invaluable help to commit things
+separately.
+On the contrary, there are commits that typically do not need to exist,
+such as commits that fix coding style or address minor review comments.
+Bear in mind that the git log is not only aimed at reviewers, but also
+at anyone who wants to understand some change you made. Do not distract
+them with cs fixes and try to make it look as if you produced perfect
+code from the start. Also, it's best if all of your commits pass the
+build, because that makes them ``git bisect`` friendly, but it also
+means they are likely to be revertable independently from other commits
```

Comment:
While it's true in theory, I can barely remember the moments when a single commit from a PR was reverted. I don't have a problem with keeping it but it doesn't seem essential. Bisect-friendly is very important, on the other hand.